### PR TITLE
Feat/2382474482 perps trade dialog adjustments

### PR DESCRIPTION
**Pull request by ShynRou**
Resolves https://sovryn.monday.com/boards/2218344956/pulses/2382474482

3 known bugs to fix:
- [x] users are sometimes asked to sign the same digest multiple times 
- [x] the open orders don't refresh automatically
- [x] the trade process doesn't seem to catch the transaction result anymore.

### CODE REVIEW

File: src/app/pages/PerpetualPage/contexts/GraphQLProvider.tsx [hunk 1]
```diff
@@ -137,11 +137,11 @@ const evaluateEndpoint = async (endpoint: GraphQLEndpoint) => {
 // TODO: add Perpetual GraphQL mainnet urls
 // fallback endpoint has to always be the last entry.
 const config: GraphQLEndpoint[] = [
-  {
-    graph:
-      'https://api.thegraph.com/subgraphs/name/omerzam/perpetual-swaps-compete',
-    index: 'https://api.thegraph.com/index-node/graphql',
-  },
+  // {
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/contexts/GraphQLProvider.tsx:140]: This is not necessary anymore, is it?

File: src/app/pages/PerpetualPage/contexts/PerpetualQueriesContext.tsx [hunk 2]
```diff
@@ -78,7 +78,7 @@ const initSocket = (
   };
 };
 
-type PerpetualValue = {
+export type PerpetualValue = {
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/contexts/PerpetualQueriesContext.tsx:81]:
```suggestion
 type PerpetualValue = {
```

File: src/app/pages/PerpetualPage/hooks/usePerpetual_OpenOrders.ts [hunk 3]
```diff
@@ -34,6 +37,23 @@ type OpenOrdersHookResult = {
 export const usePerpetual_OpenOrders = (
   address: string,
 ): OpenOrdersHookResult => {
+  const completedTransactions = usePerpetual_completedTransactions();
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/hooks/usePerpetual_OpenOrders.ts:40]: `Open Orders` table doesn't refresh automatically after new limit/stop trades so this probably doesn't work correctly.

File: src/app/pages/PerpetualPage/hooks/usePerpetual_transaction.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+  const execute = useCallback(
+    async (nonce?: number) => {
+      if (!transaction) {
+        throw Error('No transaction given to execute!');
+      }
+
+      setSending(true);
+
+      let txType: TxType = PerpetualTxMethodTypeMap[transaction.method];
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/hooks/usePerpetual_transaction.ts:76]:
```suggestion
      const txType = PerpetualTxMethodTypeMap[transaction.method];
```

File: src/app/pages/PerpetualPage/types.ts [hunk 5]
```diff
@@ -112,18 +114,19 @@ export type PerpetualTrade = {
...
-  /** wei string */
+  /** base value wei string */
   amount: string;
-  /** wei string */
+  /** limit quote price as wei string */
   limit?: string;
-  /** wei string */
-  triggerPrice?: string;
-  /** wei string */
+  /** trigger quote trigger wei string */
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/types.ts:121]:
```suggestion
  /** trigger quote price as wei string */
```

File: src/app/pages/PerpetualPage/utils/contractUtils.tsx [hunk 6]
```diff
@@ -506,3 +532,182 @@ export const balanceCallData = (
...
+  );
+
+  const deadline = Math.round(Date.now() / 1000) + 86400; // 1 day
+  const timeNow = Math.round(Date.now() / 1000);
+  const order = [
+    pair.id,
+    account,
+    floatToABK64x64(signedAmount),
+    floatToABK64x64(limitPrice),
+    0, // TODO: this is fTriggerPrice, it will need to be adjusted once we have limit orders functionality, it's 0 for market orders
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/utils/contractUtils.tsx:588]:
```suggestion
    0,
```

File: src/app/pages/PerpetualPage/utils/contractUtils.tsx [hunk 7]
```diff
@@ -506,3 +532,182 @@ export const balanceCallData = (
...
+    isClosePosition ? MASK_CLOSE_ONLY : MASK_MARKET_ORDER,
+    floatToABK64x64(leverage),
+    timeNow,
+  ];
+
+  return [order];
+};
+
+const perpetualCreateLimitTradeArgs = async (
+  context: PerpetualQueriesContextValue,
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/utils/contractUtils.tsx:600]:
```suggestion
 
```

Comment (tiltom) [reply src/app/pages/PerpetualPage/utils/contractUtils.tsx:600]: Unused arg.

File: src/app/pages/PerpetualPage/utils/contractUtils.tsx [hunk 8]
```diff
@@ -506,3 +532,182 @@ export const balanceCallData = (
...
+  });
+
+  return [order, signature];
+};
+
+const perpetualCancelLimitTradeArgs = async (
+  context: PerpetualQueriesContextValue,
+  account: string,
+  transaction: PerpetualTxCancelLimitOrder,
+): Promise<Parameters<SendTxResponseInterface['send']>[0]> => {
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/utils/contractUtils.tsx:653]:
```suggestion
const perpetualCancelLimitTradeArgs = async (
  transaction: PerpetualTxCancelLimitOrder
): Promise<Parameters<SendTxResponseInterface['send']>[0]> => {
```

Comment (tiltom) [reply src/app/pages/PerpetualPage/utils/contractUtils.tsx:653]: Unused args.

File: src/app/pages/PerpetualPage/utils/perpUtils.ts [hunk 9]
```diff
@@ -72,3 +73,74 @@ export function getRequiredMarginCollateralWithGasFees(
...
+export const MASK_LIMIT_ORDER = 0x04000000;
+
+// FIXME: move to @sovryn/perpetual-swap
+export async function createOrderDigest(
+  order: any,
+  isNewOrder: boolean,
+  managerAddress: string,
+  chainId: number,
+): Promise<string> {
+  const NAME = toUtf8Bytes('Perpetual Trade Manager');
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/utils/perpUtils.ts:92]:
```suggestion
  const name = toUtf8Bytes('Perpetual Trade Manager');
```

File: src/app/pages/PerpetualPage/utils/perpUtils.ts [hunk 10]
```diff
@@ -72,3 +73,74 @@ export function getRequiredMarginCollateralWithGasFees(
...
+  const DOMAIN_TYPEHASH = keccak256(
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/utils/perpUtils.ts:93]:
```suggestion
  const domainTypehash = keccak256(
```

File: src/app/pages/PerpetualPage/utils/perpUtils.ts [hunk 11]
```diff
@@ -72,3 +73,74 @@ export function getRequiredMarginCollateralWithGasFees(
...
+      'EIP712Domain(string name,uint256 chainId,address verifyingContract)',
+    ),
+  );
+  let domainSeparator = keccak256(
+    defaultAbiCoder.encode(
+      ['bytes32', 'bytes32', 'uint256', 'address'],
+      [DOMAIN_TYPEHASH, keccak256(NAME), chainId, managerAddress],
+    ),
+  );
+  const TRADE_ORDER_TYPEHASH = keccak256(
```
Comment (tiltom) [inline src/app/pages/PerpetualPage/utils/perpUtils.ts:104]:
```suggestion
  const tradeOrderTypehash = keccak256(
```

File: src/locales/en/translation.json [hunk 12]
```diff
@@ -2311,7 +2316,10 @@
         "liquidationPrice": "Liquidation Price:",
         "totalToReceive": "Total to Receive:",
         "maxEntryPrice": "Max Entry Price:",
-        "minEntryPrice": "Min Entry Price:"
+        "minEntryPrice": "Min Entry Price:",
+        "limitPrice": "Limit Price:",
+        "triggerPrice": "Trigger Price:",
+        "OrderSize": "Order Size:"
```
Comment (tiltom) [inline src/locales/en/translation.json:2322]:
```suggestion
        "orderSize": "Order Size:"
```

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/pages/PerpetualPage/contexts/GraphQLProvider.tsx (hunk 1 above)
Comment (ShynRou) [reply src/app/pages/PerpetualPage/contexts/GraphQLProvider.tsx:140]: Correct, sorry forgot to re-enable it.

**Review by tiltom — APPROVED** [review]

**Pull request merged**